File: aemet/Aemet-Java-IvanRC/src/main/java/dto/ListaMedicionesDto.java
package dto;

import com.squareup.moshi.Json;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Json(name = "mediciones")
@Root(name = "mediciones")
public class ListaMedicionesDto {

    @ElementList(name ="mediciones", inline = true)
    List<MedicionDto> mediciones;
}

File: aemet/Aemet-kotlin-IvanRC/src/main/kotlin/dto/ListaMedicionesDto.kt
package dto

import com.squareup.moshi.Json
import org.simpleframework.xml.ElementList


data class ListaMedicionesDto (
    @Json(name = "mediciones")
    @field:ElementList(name = "mediciones", inline = true)
    @param:ElementList(name = "mediciones", inline = true)
    val mediciones: List<MedicionDto>
)